memgpt/schemas/block.py

```diff
@@ -0,0 +1,51 @@
+from typing import List, Optional, Union
+
+from pydantic import BaseModel, Field, validator
+
+# block of the LLM context
+
+
+class Block(BaseModel):
+    """Block of the LLM context"""
+
+    description: Optional[str] = Field(None, description="Description of the block.")
+    limit: int = Field(2000, description="Character limit of the block.")
+    value: Optional[Union[List[str], str]] = Field(None, description="Value of the block.")
+
+    def __setattr__(self, name, value):
+        """Run validation if self.value is updated"""
+        super().__setattr__(name, value)
+        if name == "value":
+            # run validation
+            self.__class__.validate(self.dict(exclude_unset=True))
+
+    @validator("value", always=True)
+    def check_value_length(cls, v, values):
```

Comment:
might want to do this as a [model validator set to "after"](https://docs.pydantic.dev/latest/api/functional_validators/#pydantic.functional_validators.model_validator) - that gets rid of this magic 2000 value. 
so basically
1. limit gets set and defaults to 2000 per the attribute
2. value is checked against limit, which is now the given value or 2k

otherwise you get into this icky spot where a developer changes the default on limit to 4k, but the validator is failing on 2500, and confusion ensues

Reply:
ah ok I think I tried to do this before, but didn't realize `validate_assignment=True` existed so it didn't update after I changed the values after the instance was created